bitnami/postgresql-repmgr/16/debian-12/rootfs/opt/bitnami/scripts/librepmgr.sh

```diff
@@ -625,15 +625,25 @@ repmgr_clone_primary() {
         rm -rf "$waldir" && ensure_dir_exists "$waldir"
     fi
 
+    if am_i_root; then
+        chown -R "$POSTGRESQL_DAEMON_USER:$POSTGRESQL_DAEMON_GROUP" "$POSTGRESQL_DATA_DIR"
+    fi
+
     info "Cloning data from primary node..."
     local -r flags=("-f" "$REPMGR_CONF_FILE" "-h" "$REPMGR_CURRENT_PRIMARY_HOST" "-p" "$REPMGR_CURRENT_PRIMARY_PORT" "-U" "$REPMGR_USERNAME" "-d" "dbname=$REPMGR_DATABASE host=$REPMGR_CURRENT_PRIMARY_HOST connect_timeout=$REPMGR_CONNECT_TIMEOUT" "-D" "$POSTGRESQL_DATA_DIR" "standby" "clone" "--fast-checkpoint" "--force")
+    local repmgr_cmd=()
+
+    if am_i_root; then
+        repmgr_cmd+=("run_as_user" "$POSTGRESQL_DAEMON_USER")
+    fi
+
+    repmgr_cmd+=("${REPMGR_BIN_DIR}/repmgr")
```

Comment:
Since this block is repeated in quite a few places, does it make sense to create it its own function?

Reply:
Definitely!  Where do you suggest implementing the function?  Maybe somewhere in the prebuildfs scripts?

Reply:
I think it will be fine to define it in this very same library file `librepmgr.sh`. Maybe something like:

```
repmgr_execute() {
    if am_i_root; then
        debug_execute run_as_user "$POSTGRESQL_DAEMON_USER" "${REPMGR_BIN_DIR}/repmgr" "$@"
    else
...
}
```

Or with some local variable to help with the command args too